src/Model/Configuration.py

```diff
@@ -0,0 +1,121 @@
+
+import platform
+import ctypes
+import sqlite3
+import os
+import functools
+
+from src.Model.Singleton import Singleton
+
+
+def set_up_hidden_dir():
+    """
+    Set up the hidden directory
+    """
+    directory_name = '.OnkoDICOM'
+
+    # Set USER_HIDDEN and USER_HOME env variable
+    os.environ['USER_HOME'] = os.path.expanduser("~")
+    if platform.system() == 'Windows':
+        path = os.environ.get('USER_HOME') + "\\" + directory_name + "\\"
+    else:
+        path = os.environ.get('USER_HOME') + "/" + directory_name + "/"
+    os.environ['USER_HIDDEN'] = path
+
+    # Create and hide the hidden directory
+    if not os.path.exists(path):
+        os.mkdir(path)
+        if platform.system() == 'Windows':
+            # Hide the directory for Windows machines
+            ctypes.windll.kernel32.SetFileAttributesW(path, 2)
+
+
+class SqlError(Exception):
+    pass
+
+
+def error_handling(function):
+    @functools.wraps(function)
+    def wrapper(*args):
+        try:
+            result = function(*args)
+            return result
+        except (sqlite3.Error, sqlite3.OperationalError):
+            try:
+                if os.path.exists(os.environ['USER_HIDDEN']):
+                    os.remove(os.environ['USER_HIDDEN'])
+                set_up_hidden_dir()
+                raise SqlError()
+            except OSError:
+                raise SqlError()
+    return wrapper
+
+
+class Configuration(metaclass=Singleton):
+    """
+        This Singleton class represents the user configuration. It contains the default directory for the DICOM files.
+        The object will contain the connection to a sqlite database set below, and the user can update the default
```

Comment:
"The object will contain the connection to an SQLite database set below. The user will be able to update the default directory during runtime."